huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset.go

```diff
@@ -0,0 +1,612 @@
+		ImportData:        true,
+		ImportAnnotations: utils.Bool(d.Get("import_labeled_enabled").(bool)),
+		Schema:            schemas,
+		Labels:            labels,
+	}
+
+	if format := buildLabelFormatParamter(d); format != nil {
+		rst.LabelFormat = *format
+	}
+
+	return &rst, nil
+}
+
+func buildLabelFormatParamter(d *schema.ResourceData) *dataset.LabelFormat {
+	if v, ok := d.GetOk("label_format"); ok {
+		configRaw := v.([]interface{})[0].(map[string]interface{})
+		configs := dataset.LabelFormat{
+			LabelType:           configRaw["type"].(string),
+			TextLabelSeparator:  configRaw["text_label_separator"].(string),
+			TextSampleSeparator: configRaw["label_separator"].(string),
+		}
+		return &configs
+	}
+	return nil
+}
+
+func buildDataSourceParamter(d *schema.ResourceData) (dataSources []dataset.DataSource, e error) {
+	item := d.Get("data_source").([]interface{})[0].(map[string]interface{})
+
+	dataType := item["data_type"].(int)
+	dataSource := dataset.DataSource{
+		DataType:         dataType,
+		WithColumnHeader: utils.Bool(item["with_column_header"].(bool)),
+	}
+
+	path := item["path"].(string)
+	if dataType == 0 {
+		if path == "" {
+			e = fmtp.Errorf("When import data from OBS, path is required.")
+			return
+		}
+		dataSource.DataPath = path
+	}
+
+	clusterId := item["cluster_id"].(string)
+	databaseName := item["database_name"].(string)
+	tableName := item["table_name"].(string)
+	userName := item["user_name"].(string)
+	password := item["password"].(string)
+
+	if dataType == 1 {
+		if clusterId == "" || databaseName == "" || tableName == "" || userName == "" || password == "" {
+			e = fmtp.Errorf("When import data from DWS, cluster_id, database_name, table_name, user_name and" +
+				" password are required.")
+			return
+		}
+
+		dataSource.SourceInfo.ClusterId = clusterId
+		dataSource.SourceInfo.DatabaseName = databaseName
+		dataSource.SourceInfo.TableName = tableName
+		dataSource.SourceInfo.UserName = userName
+		dataSource.SourceInfo.UserPassword = password
+	}
+
+	queueName := item["queue_name"].(string)
+
+	if dataType == 2 {
+		if queueName == "" || databaseName == "" || tableName == "" {
+			e = fmtp.Errorf("When import data from DLI, queue_name, database_name and table_name are required.")
+			return
+		}
+
+		dataSource.SourceInfo.QueueName = queueName
+		dataSource.SourceInfo.DatabaseName = databaseName
+		dataSource.SourceInfo.TableName = tableName
+	}
+
+	if dataType == 4 {
+		if clusterId == "" || path == "" {
+			e = fmtp.Errorf("When import data from MRS, cluster_id and path are required.")
+			return
+		}
+
+		dataSource.SourceInfo.ClusterId = clusterId
+		dataSource.SourceInfo.Input = path
+	}
+
+	dataSources = append(dataSources, dataSource)
+	return
+}
+
+func buildLabelsParamter(v interface{}) (labels []dataset.Label, e error) {
+	if v != nil {
+		configRaw := v.([]interface{})
+		for _, item := range configRaw {
+			tmp := item.(map[string]interface{})
+			labels = append(labels, dataset.Label{
```

Comment:
It is recommended that this function does not use an explicit declaration of the return value, because it will waste the memories.
```
func buildLabelsParamter(v interface{}) ([]dataset.Label, error) {
	if v != nil {
		configRaw := v.([]interface{})
		labels := make([]dataset.Label, len(configRaw))
		for i, item := range configRaw {
			tmp := item.(map[string]interface{})
			labels[i] = dataset.Label{
```
Same problems exists elsewhere.

Reply:
unnecessary! 
just left one reference when throwing an error.